### ISSUE TITLE
Reading messages in Chronicle Queue Tailer (v5.16.11) does not auto-move the index to next roll cycle

### ISSUE THREAD

**Issue by samratbatth** [OpenHFT/Chronicle-Queue#537]
The tailer does not auto-roll to next cycle in case of missing cycles.

1. **Setup / Libs:**
jvm openjdk 11, Ubuntu 16.04, openhft.affinity/3.1.9, chronicle-map/3.16.0, chronicle-queue/5.16.11, chronicle-bytes/1.16.23, chronicle-core/1.16.20, chronicle-wire/1.16.16, chronicle-threads/1.16.3, jna/4.4.0

2. **Steps to replicate the issue:**
- Start WriterProcess - let it finish.
- Start ReaderProcess - see the 5 print statements.
- Stop ReaderProcess
- Wait for some time - 10 mins.
- Start WriterProcess again - let it finish or keep running this process.
- Start ReaderProcess - it prints only the first 5 print statements and nothing prints after this. Even if the WriterProcess is running/writing to queue the tailer in this process does not move forward.

 ```
public class WriterProcess {
        public static void main(String[] args) throws InterruptedException {
            final String dir = "/tmp/demo/";
            final LocalTime localTime = LocalTime.of(17, 0);
            final ZoneId zoneID = ZoneId.of("America/New_York");
            final ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(2);
            final SingleChronicleQueue queue  = SingleChronicleQueueBuilder.binary(dir)
                    .blockSize((long) Math.pow(2, 23))
                    .rollCycle(RollCycles.MINUTELY)
                    .rollTime(localTime, zoneID)
                    .build();
            final ExcerptAppender appender = queue.acquireAppender();
            // pre touch
            scheduledExecutorService.scheduleAtFixedRate(appender::pretouch,0,30, TimeUnit.SECONDS);
            // write data
            System.out.println("writing data ...");
            writeData(appender, 5);
            // close queue
            System.out.println("shutting down now ...");
            queue.close();
            scheduledExecutorService.shutdown();
            scheduledExecutorService.awaitTermination(1, TimeUnit.SECONDS);
        }
        public static void writeData(ExcerptAppender appender, int count) {
            int ctr = 0;
            String dateStr;
            Date date = new Date();
            while (true) {
                dateStr = date.toString();
                appender.writeText("["+ctr+"] Written " + dateStr);
                System.out.println("["+ctr+"] Written " + dateStr);
                ctr++;
                if (ctr >= count) {
                    break;
                }
                try {
                    Thread.sleep(65_000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
        }
    }
    public class ReaderProcess {
        public static void main(String[] args) {
            final String dir = "/tmp/demo/";
            final LocalTime localTime = LocalTime.of(17, 0);
            final ZoneId zoneID = ZoneId.of("America/New_York");
            final SingleChronicleQueue queue  = SingleChronicleQueueBuilder.binary(dir)
                    .blockSize((long) Math.pow(2, 23))
                    .rollCycle(RollCycles.MINUTELY)
                    .rollTime(localTime, zoneID)
                    .build();
            final ExcerptTailer tailer = queue.createTailer();
            tailer.toStart();
            // read data
            System.out.println("reading data ...");
            readData(tailer, 25);
            // close
            System.out.println("shutting down now ...");
            queue.close();
        }
        public static void readData(ExcerptTailer tailer, int count) {
            int ctr = 0;
            Bytes data = Bytes.allocateDirect(new byte[500]);
            while (true) {
                if (tailer.readBytes(data)) {
                    System.out.println("["+ctr+"] Read {"+ data + "}");
                    ctr++;
                    if (ctr >= count) {
                        break;
                    }
                }
            }
        }
    } 
```

**Reference:** https://stackoverflow.com/questions/52783626/reading-messages-in-chronicle-queue-tailer-v5-16-11-does-not-auto-move-the-ind

**Comment by dpisklov** [comment]
Closing this as Peter has already replied on stackoverflow with correct way of doing that

**Comment by hft-team-city** [comment]
Released in Chronicle-Queue-5.17.0 (https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.0), BOM-2.17.30 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.30)
